TradeItIosTicketSDK2/TradeItOrdersTableViewManager.swift

```diff
@@ -0,0 +1,240 @@
+    // MARK: UITableViewDataSource
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        return self.orderSectionPresenters[indexPath.section].cell(forTableView: tableView, andRow: indexPath.row)
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        guard let orderSectionPresenter = self.orderSectionPresenters[safe: section] else { return 0 }
+        return orderSectionPresenter.numberOfRows()
+    }
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        if self.orderSectionPresenters.isEmpty {
+            self.ordersTable?.backgroundView = noResultsBackgroundView
+        } else {
+            self.ordersTable?.backgroundView = nil
+        }
+        return self.orderSectionPresenters.count
+    }
+    
+    func tableView(_ tableView: UITableView, estimatedHeightForRowAt indexPath: IndexPath) -> CGFloat {
+        return CGFloat(TradeItOrdersTableViewManager.ORDER_CELL_HEIGHT)
+    }
+    
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return CGFloat(TradeItOrdersTableViewManager.ORDER_CELL_HEIGHT)
+    }
+    
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        return self.orderSectionPresenters[section].header(forTableView: tableView)
+    }
+    
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return self.orderSectionPresenters[section].heightForHeaderInSection()
+    }
+    
+    func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        return CGFloat.leastNormalMagnitude
+    }
+    
+    // MARK: Private
+    
+    private func addRefreshControl(toTableView tableView: UITableView) {
+        let refreshControl = UIRefreshControl()
+        refreshControl.attributedTitle = NSAttributedString(string: "Refreshing...")
+        refreshControl.addTarget(
+            self,
+            action: #selector(initiateRefresh),
+            for: UIControlEvents.valueChanged
+        )
+        TradeItThemeConfigurator.configure(view: refreshControl)
+        tableView.addSubview(refreshControl)
+        self.refreshControl = refreshControl
+    }
+    
+    /**
+     * This is to split orders in order to have a specific section for group orders
+    **/
+    private func getSplittedOrdersArray(orders: [TradeItOrderStatusDetails]) -> [[TradeItOrderStatusDetails]]{
+        return orders.reduce([[]], { splittedArrays, order in
+            var splittedArraysTmp = splittedArrays
+            let lastResult: [TradeItOrderStatusDetails] = splittedArrays[(splittedArrays.endIndex - 1)]
+            
+            if !order.isGroupOrder() && !lastResult.contains(order) { // this is not a group order, we can append the order
+                splittedArraysTmp[(splittedArraysTmp.endIndex - 1)].append(order)
```

Comment:
`.last`

Reply:
Actually last return an immutable variable, so ?I can't use it here, but I will for above